lib/options.nix

```diff
@@ -466,6 +467,57 @@ rec {
       ${concatMapStringsSep "\n" (defFile: "  - ${defFile}") opt.files}
     '';
 
+  /**
+    Filters an options tree.
+
+
+    # Type
+
+
+    ```
+    filterOptions :: ([n] -> o -> bool) -> opts -> opts
+    ```
+
+
+    # Examples
+    :::{.example}
+    ## `filterOptions` usage example
+
+
+    ```nix
+    filterOptions
+      (path: option: hasPrefix "a-" option.description || elem "services" path)
+      {
+        services.foo = mkOption { description = "foo service"; };
+        programs.a-prog = mkOption { description = "a program"; };
+        programs.b-prog = mkOption { description = "b program"; };
+      }
+    => {
+      services.foo = mkOption { description = "foo"; };
+      programs.a-prog = mkOption { description = "program"; };
+    }
```

Comment:
This can not be the result. Please test it.

Reply:
Amended.